### PR TITLE
Ea 2911 zoho official language field

### PR DESCRIPTION
**Pull request by SrdjanStevanetic**

### CODE REVIEW

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/Organization.java [hunk 1]
```diff
@@ -52,7 +53,8 @@
   FOAF_MBOX,
   HAS_ADDRESS,
   IDENTIFIER,
-  SAME_AS
+  SAME_AS,
+  EDM_LANGUAGE
```
Comment (gsergiu) [inline entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/Organization.java]: we should use the officialLanguage in the data model and only use "language" in the searialization (keeping consistent the naming in the xml and json serialization)

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/vocabulary/WebEntityFields.java [hunk 2]
```diff
@@ -94,6 +94,7 @@
   public static final String FOAF_HOMEPAGE = "homepage";
   public static final String FOAF_PHONE = "phone";
   public static final String FOAF_MBOX = "mbox";
+  public static final String EDM_LANGUAGE = "edmLanguage";
```
Comment (gsergiu) [inline entity-management-definitions/src/main/java/eu/europeana/entitymanagement/vocabulary/WebEntityFields.java]: this should be renamed to language

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/web/xml/model/XmlConstants.java [hunk 3]
```diff
@@ -86,6 +86,7 @@
   public static final String XML_PHONE = "phone";
   public static final String XML_MBOX = "mbox";
   public static final String XML_HAS_ADDRESS = "hasAddress";
+  public static final String XML_LANGUAGE = "language";
```
Comment (gsergiu) [inline entity-management-definitions/src/main/java/eu/europeana/entitymanagement/web/xml/model/XmlConstants.java:89]: this is an xml fieldname, therefore it should be placed in parent interface - WebEntityFields

File: entity-management-tests/src/integration-test/java/eu/europeana/entitymanagement/testutils/IntegrationTestUtils.java [hunk 4]
```diff
@@ -213,8 +213,8 @@
           ORGANIZATION_NATURALIS_ZOHO_RESPONSE,
           ORGANIZATION_GFM_URI_ZOHO,
           ORGANIZATION_GFM_ZOHO_RESPONSE,
-          ORGANIZATION_PCCE_URI_ZOHO,
-          ORGANIZATION_PCCE_ZOHO_RESPONSE,
+          ORGANIZATION_BYZART_URI_ZOHO,
```
Comment (gsergiu) [inline entity-management-tests/src/integration-test/java/eu/europeana/entitymanagement/testutils/IntegrationTestUtils.java]: please add back the PCCE organization as it is used in integration tests

File: entity-management-tests/src/integration-test/java/eu/europeana/entitymanagement/web/EntityRegistrationIT.java [hunk 5]
```diff
@@ -319,16 +319,17 @@ public void registerZohoOrganizationBergerShouldBeSuccessful() throws Exception
   }
 
   @Test
-  void registerOrganizationWithoutWikidataSameAsShouldBeSuccessful() throws Exception {
+  void registerOrganizationWithoutWikidataSameAsAndWithEdmLanguageShouldBeSuccessful()
```
Comment (gsergiu) [inline entity-management-tests/src/integration-test/java/eu/europeana/entitymanagement/web/EntityRegistrationIT.java]: please revert back this integration test and create a new one for byzart Organization

File: entity-management-tests/src/integration-test/resources/content/organization_register_zoho_byzart.json [hunk 6]
```diff
@@ -1,4 +1,4 @@
 {
   "type": "Organization",
-  "id": "https://crm.zoho.com/crm/org51823723/tab/Accounts/1482250000000338555"
+  "id": "https://crm.zoho.com/crm/org51823723/tab/Accounts/1482250000042591010"
```
Comment (gsergiu) [inline entity-management-tests/src/integration-test/resources/content/organization_register_zoho_byzart.json]: please revert back the deletion of PCCE json file

File: entity-management-zoho/src/main/java/eu/europeana/entitymanagement/zoho/organization/ZohoOrganizationConverter.java [hunk 7]
```diff
@@ -85,6 +85,10 @@ public static Organization convertToOrganizationEntity(Record zohoRecord) {
       org.setAddress(address);
     }
 
+    List<String> edmLanguage =
+        ZohoUtils.stringListSupplier(zohoRecord.getKeyValue(ZohoConstants.OFFICIAL_LANGUAGE_FIELD));
+    if (!edmLanguage.isEmpty()) org.setEdmLanguage(edmLanguage);
```
Comment (gsergiu) [inline entity-management-zoho/src/main/java/eu/europeana/entitymanagement/zoho/organization/ZohoOrganizationConverter.java]: please use the {} for the body of the if statement, as it breaked the sonar cube rules

**Review by gsergiu — CHANGES_REQUESTED** [review]
the previously created integration tests must not be removed, the naming of the new field should be improved

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/Organization.java (hunk 1 above)
Comment (SrdjanStevanetic) [reply entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/Organization.java]: Renamed to LANGUAGE as agreed.

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/vocabulary/WebEntityFields.java (hunk 2 above)
Comment (SrdjanStevanetic) [reply entity-management-definitions/src/main/java/eu/europeana/entitymanagement/vocabulary/WebEntityFields.java]: Renamed to language as agreed.

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/web/xml/model/XmlConstants.java (hunk 3 above)
Comment (SrdjanStevanetic) [reply entity-management-definitions/src/main/java/eu/europeana/entitymanagement/web/xml/model/XmlConstants.java:89]: The names for all other fields are placed here, I am not sure what you exactly mean.

File: entity-management-tests/src/integration-test/java/eu/europeana/entitymanagement/testutils/IntegrationTestUtils.java (hunk 4 above)
Comment (SrdjanStevanetic) [reply entity-management-tests/src/integration-test/java/eu/europeana/entitymanagement/testutils/IntegrationTestUtils.java]: Added back.

File: entity-management-tests/src/integration-test/java/eu/europeana/entitymanagement/web/EntityRegistrationIT.java (hunk 5 above)
Comment (SrdjanStevanetic) [reply entity-management-tests/src/integration-test/java/eu/europeana/entitymanagement/web/EntityRegistrationIT.java]: Done.

File: entity-management-tests/src/integration-test/resources/content/organization_register_zoho_byzart.json (hunk 6 above)
Comment (SrdjanStevanetic) [reply entity-management-tests/src/integration-test/resources/content/organization_register_zoho_byzart.json]: Done.

File: entity-management-zoho/src/main/java/eu/europeana/entitymanagement/zoho/organization/ZohoOrganizationConverter.java (hunk 7 above)
Comment (SrdjanStevanetic) [reply entity-management-zoho/src/main/java/eu/europeana/entitymanagement/zoho/organization/ZohoOrganizationConverter.java]: Done.

**Pull request merged**